pkg/iniparser_test.go

```diff
@@ -0,0 +1,1308 @@
+package iniparser
+
+import (
+	"reflect"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type LoadFromStringTestCase struct {
+	desc          string
+	input         string
+	expectedError error
+	expectedMap   map[string]map[string]string
+}
+
+type LoadFromFileTestCase struct {
+	desc          string
+	path          string
+	expectedError error
+	expectedMap   map[string]map[string]string
+}
+
+type StringTestCase struct {
+	desc     string
+	inputMap map[string]map[string]string
+}
+
+type SaveToFileTestCase struct {
+	desc          string
+	inputPath     string
+	inputMap      map[string]map[string]string
+	expectedError error
+}
+
+type GetSectionNamesTestCase struct {
+	desc        string
+	inputMap    map[string]map[string]string
+	outputSlice []string
+}
+
+type GetSectionsTestCase struct {
+	desc           string
+	inputOutputMap map[string]map[string]string
+}
+
+type GetSetTestCase struct {
+	desc          string
+	inputMap      map[string]map[string]string
+	section       string
+	key           string
+	value         string
+	expectedError error
+}
+
+func sliceEqual(a, b []string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	a_copy := make([]string, len(a))
+	b_copy := make([]string, len(b))
+
+	copy(a_copy, a)
+	copy(b_copy, b)
+
+	sort.Strings(a_copy)
+	sort.Strings(b_copy)
+
+	return reflect.DeepEqual(a_copy, b_copy)
+}
+
+func TestINI_LoadFromString(t *testing.T) {
```

Comment:
go functions names shouldn't contain '_', a test for a function called "Xxx" should be named "TestXxx", "Test" then the function name